Packs/VMRay/Integrations/VMRay/VMRay.py

```diff
@@ -142,10 +148,40 @@ def find_error(may_be_error_inside):
                     return err_r
         return None
 
+    def reset_file_buffer(files):
+        """ The requests library reads the file buffer to the end.
+            If we want to try to submit again, we need to set the file buffer to
+            beginning manually.
+        Args:
+            files: (dict)
+        """
+
+        if not isinstance(files, dict):
+            return
+
+        try:
+            fobj = files["sample_file"][1]
+            fobj.seek(0, 0)
+        except (IndexError, KeyError):
+            return
+
     url = SERVER + url_suffix
     r = requests.request(
         method, url, params=params, headers=HEADERS, files=files, verify=USE_SSL, proxies=PROXIES
     )
+
+    if RETRY_ON_RATE_LIMIT and (r.status_code == RATE_LIMIT_REACHED or "Retry-After" in r.headers):
+        seconds_to_wait = random.uniform(1.0, 5.0)
+        try:
+            seconds_to_wait += float(r.headers.get("Retry-After", 0))
+        except ValueError:
+            pass
+
+        demisto.debug(f"Rate limit exceeded! Waiting {seconds_to_wait} seconds and then retry.")
+        time.sleep(seconds_to_wait)  # pylint: disable=sleep-exists
+        reset_file_buffer(files)
+        return http_request(method, url_suffix, params, files, ignore_errors, get_raw)
```

Comment:
In this case, it would be better to assign the response to a variable and then continue with the logic of the function. 
```suggestion
        r = http_request(...)
```

Why are the request args different from the one above? 
```suggestion
        r = requests.request(
        method, url, params=params, headers=HEADERS, files=files, verify=USE_SSL, proxies=PROXIES
    )
```

Reply:
The arguments differ, because it is a recursive function call and not a direct request. We need to call `http_request` recursively here, since it is possible that retrying only once could end up in a rate limit that was triggered by another waiting request.

Reply:
Missed the recursion. So what is the stop condition? How do you make sure you won't enter an infinite loop and get a stack overflow? you should count the number of tries.